### DISCUSSION TITLE
Add other legal disclaimer

### DISCUSSION THREAD

**Discussion by sguinetti** [iptv-org/community#102] [Ideas]
Hello and thanks for accepting several PR to include Peruvian television in this repository. Now, I have a suggestion and that is to add an additional note below the legal DMCA disclaimer.

> This repository offers the widest public streaming coverage to full audiences around the world. However, each country establishes limits on their freedom of expression for mass media in certain circumstances, such as child privacy rights, moral rights, fraud prevention or other ethical principles. So if the channel broadcasts content that violates on your personality, moral or privacy rights or this infringes acceptable content during children's programming hours under domestic laws, **you should contact the corresponding regulatory committee** from official website to report the issue and find a solution. If the channel does not have one or no satisfactory response from regulatory committee was obtained, you should contact the competent authority from their country of origin. Anyway, with special emphasis on online-only channels, **we cannot guarantee the compliance of local channel broadcasting under national laws and we are not responsible for any unexpected unlawful consequence when broadcasting media outside their country of origin**.
>
> Channel stream information, as well as channel names, logos, countries of origin and EPGs may be inaccurate or outdated. If you need to add updated information on the user-submitted links, we would be grateful to report to us or create a PR making the changes.

Second version (more simple and USA-compatible):

> This repository offers the broadest public streaming coverage to complete audiences worldwide. However, some countries may have laws that more or less restrict the broadcasting of content for dishonest purposes, such as overexposure of minors (in US is under COPPA act), harm to moral rights (nonexistent in DCMA law) or terrorism (PATRIOT ACT). Because we do not store the files, however, the stream may facilitate the dissemination of content to harm the image of you or your family that we may not be able to control. In case this happens, we suggest contacting the corresponding regulatory commission from the official website to report the problem and prevent future incidents of this type. We may remove stream urls if an authority orders the takedown of stream url(s) for abuse of broadcasting that may violate federal laws.

This may help in certain transmissions that may publish sensitive and/or private content that could interfere with the laws of other countries. As well as inaccurate information in sometimes.

**Comment by BellezaEmporium** [comment]
Sounds good to me.

**Reply by sguinetti** [reply]
I have already made some additional updates. I think it's ready.

**Reply by dct-infra** [reply]
> we cannot guarantee the compliance of local channel broadcasting under national laws and we are not responsible for any unexpected unlawful consequence when broadcasting media outside their country of origin.

> This may help in certain transmissions that may publish sensitive and/or private content that could interfere with the laws of other countries.

Sound good but doesn't apply here I guess, simply because this repo doesn't HOST nor streams any content.
This text seems to imply that we are declining a responsibility that could, under certain circumstances, be ours, while we don't have any regarding CONTENT.
Let's not shoot ourselves in the foot, we are not concerned with any compliance whatsoever regarding the broadcasted CONTENT in relation to any laws/jurisdiction in the world except for DMCA or maybe EU copyright laws depending on Github's country(ies) of registration.

**Reply by sguinetti** [reply]
The "we cannot guarantee the compliance of local channel broadcasting under national laws and we are not responsible for any unexpected unlawful consequence when broadcasting media outside their country of origin" phase could be deleted so that we don't shoot ourselves in the foot.

But the point I am making is about retransmiting content and it's unexpectedly problematic. Let's think for example that there is a user watching a stream from our list (IPTV repo in a IPTV app), maybe the stream transmits sensitive data of a children. Suddenly someone notices this and someone wants to remove that stream from our list because we "facilitate" the broadcasting of private content (maybe because several people watch that stream because it is easier to use this list than because of the website).

> Let's not shoot ourselves in the foot, we are not concerned with any compliance whatsoever regarding the broadcasted CONTENT in relation to any laws/jurisdiction in the world except for DMCA or maybe EU copyright laws depending on Github's country(ies) of registration.



In the United States (Github is from this country) there is not only the DCMA law there is also the COPPA law (https://en.wikipedia.org/wiki/Children%27s_Online_Privacy_Protection_Act, it's like GDPR), and they can request the closure of the channel under the second law. What do you think?

Anyway, I do not consider myself an American to know the removal situation more as DCMA. However this disclaimer may help a little to be aware of the dissemination of content that may be legal in one country and illegal in another. Thank you for your feedback.

**Reply by dct-infra** [reply]
_DISCLAIMER_
Get a coffee or some energy drink, this will be looong :)
At the end of the post you'll be fed up with the word CONTENT, it's intentional.
_END OF DISCLAIMER_

@sguinetti - The topic is very interesting. I understand your point and there is always a need to optimize some legal aspects whenever making something publicly available. However, in order to do this the best way possible we need to clarify things and be more precise.

In 3 points.
First off, we're not retransmitting anything, we just keep fresh lists of public freely acessible streams that aren't private in any way. We do not provide CONTENT.
We don't even "facilitate" access, such words are delicate since facilitating implies we're making something supposedly hard become easy. It also implies being complicit of any wrong doing, knowingly or not. 

The fact is that we just keep dynamic (see why dynamic below) and categorized lists of channels in a specific format (m3u), making sure, in good faith, that CONTENT is as appropriate as possible and geo-blocks respected. This, is done at addition time only, no further CONTENT monitoring is made, only automated stream availability checks. 
In some cases we provide "a different access method" a.k.a a stack of clever opensource stuff we call "proxy" which is in fact not a proxy per se, technically/legally speaking, it's more of an URL grabber used by devices that can't translate a certain type of stream urls correctly. Still, no CONTENT or retransmission involved.
Sometimes we stumble on existing urls where channel names change, CONTENT changes and even country of origin changes. Listings and CONTENT are dynamic by nature, we canot endorse nor decline CONTENT according to any local national law, except, again as for copyright laws. We simply have no control over CONTENT. We might if access to our lists was provided as a service with contractual stuff, turnover and a few hundred employees behind it. By the way, any investors interested? E-mail me! 
The fact is that even though categorized, these lists are still dynamic raw material a user must chose from under his sole responsibility according to his standards and the laws of his country of residence. Obvious, no disclaimer needed.

Second, COPPA is not quite the right example since it does not specifically deal with CONTENT, it's mainly about PII collection from children under the age of 13, it has been tailored for social media and other free/paid services where subscriptions and PII are necessary. We don't deal with PII here.

Last, I'll go further with your example about child safety. Now what about the adult channel list? Maybe that's what you're referring to? If it's the case, then we agree that this specific list should be secured and users verified in a way somehow. But still, in my opinion every parent is responsible for what CONTENT is accessed by his child online. Again, obvious, no disclaimer needed

I'm using music and entrertainment lists myself for my family. According to my standards, some channels on those lists are absolutely a no go. should I make a PR and delete them or just make sure they get off the list automatically on every refresh? Obviously, second option makes more sense: my family, their safety, my responsibilty.

So unless we have some big lawyers in here willing to assist I don't think we should get into complex and lenghy disclaimers that may bring "unexpectedly problematic" head aches. (Not making fun @sguinetti I really like the wording!).
Why not just chose a license? Public domain? Anything else? Ideas people?

**Reply by sguinetti** [reply]
Anyway I found it funny that you are diverting to the CONTENT of the channel, which as I said in the proposal it follows that the "content is the responsibility of the one who produces the content". You are right about the word "streaming" or "broadcasting", because the list maintainers don't broadcasting, as the closest thing would be for the maintainers to offer a "proxy" platform to stream url. And yes it is in the public domain, but if something a crime and I want to dedicate to the public domain, then will I cease to be the author of it?

I am not limiting myself to a specific point in this discussion about any problematic effects that may occur. Perhaps "de minimis" events are not problematic for all, now let's see if they occur with more serious effects.

I know a little of the US the laws are quite different from EU or other countries perhaps because of amendments to the constitution. I'm going  beyond and I try to be the bad guy here (or the devil's advocate, even if I am not) and would like to at least challenge you as to how I would solve this example. A random person (no we) created a web URL for their plan. This person (cracker?) sees the IPTV repository site that plans a url that will broadcast non-harmful content until the PR is accepted. A few days later that person changes the broadcast and after a clever timing, broadcasts child pornography. There is no removal request by DCMA, because they already know that the person is not interested in removing that pornography. Also the reviewers did not notice. To what extent could we be responsible for keeping the url that transmits child pornography without a definite time until removed? Can we be responsible for the dynamic change of content (I read your opinion again and I change the word CONTENT to CP CONTENT and maybe you're still right because this bad guy example looks like a fallacy, but in some sentences it seems a bit shocking)? This example not only infringes on privacy and moral rights of the victims. Remember that this is only an assumption that may not exist for now but reflects the points of the proposal.

I will admit that a specialized lawyer (no me) would have a better solution to the problem and that this would already be too controversial to write about because this example is not funny.

**Reply by dct-infra** [reply]
@sguinetti  It's always a good thing to be the devil's advocate and you're right, best way to bring different view points to the table.
`Can we be responsible for the dynamic change of CONTENT?` Better: can we control it in the first place? Certainly not, we maintain a list, still in good faith, but shit can happen! Implicit, because we're a public list. CONTENT is completely out of our control. We might be responsible if it's PROVEN that we knew of unlawful actions and maintained the listing, something that will never happen. Firstly because we would drop the link voluntarily right away if found and secondly because we would do the same upon notice.
The rule in legal matters is that we cannot emphasize on everything, things can get back to us like a boomerang if we don't master them. The implicit nature of things is sufficient. We're just a list!
An by the way I've just found the lists were on the Public Domain licence, perfect!

**Comment by kawaii-ghost** [comment]
Hello, i have a question.

Yesterday I found some illegal iptv link and I can prove evidence to show that's illegal. But I'm not the copyright holder.

Could everybody suggest to remove copyright iptv link or only the copyright holder could suggest to remove it ?

**Reply by freearhey** [reply]
This repository contains only publicly available links to channel broadcasts.

If you find a stream that you don't think should be available to the public it's better contact the copyright holder to have the source removed.
